### ISSUE TITLE
LibC in dependancy

### ISSUE THREAD

**Issue by darnuria** [mongodb/bson-rust#201]
Hello,

I watched the conference of Irina Shestak: https://www.youtube.com/watch?v=lLzFJenzBng

So I wondered myself why this crate is using `libc`? So I tried to compile it without it, it turned out it worked and also turned out it passed all the test with `cargo test` and

So I searching "why libc is in deps"? 

- I ran: `ag libc` found just one occurence in `Cargo.toml`
- I searched down in git history why it's just the Cargo file `git log -Glibc`
- It turn out it was introduced in your tree in commit: 4110055abaf98331cb8484a9826e14ac6b8326f8.
- But some related code was removed in: 704b41da8ac4b746066ef3af4846ea4b5685e437
- Was updated to 0.2 in be922fa43671000f95466ebd557349b4aa0b19cb
- And usage was removed in d8485d67cf40d683fd7a090cee33a721fa577688

In short somebody have done all the work except removing the direct dependency in this crate.
Maybe it's time to remove this libc entry in `Cargo.toml` dependency.

Linked PR: It turn out that a PR is already opened here https://github.com/mongodb/bson-rust/pull/137. (github search did not included that)

If it's Okay to make a mini-PR with just removing `libc` entry tell me.

PS: I am wrong please feel free to close this issue.
Edit: just saw that somebody made a PR about it.

**Comment by saghm** [comment]
We're planning on merging #137 soon; we just need to merge #200 first to fix CI for it.

**Comment by saghm** [comment]
#137 is now merged, so the library no longer depends on libc
